Tiqr/Classes/SecretService.m

```diff
@@ -137,13 +137,32 @@ - (BOOL)updateOrStoreSecret:(NSData *)secret forIdentity:(Identity *)identity {
 }
 
 - (BOOL)deleteSecretForIdentityIdentifier:(NSString *)identityIdentifier providerIdentifier:(NSString *)providerIdentifier; {
-    NSMutableDictionary *query = [[NSMutableDictionary alloc] init];
-    query[(__bridge id)kSecClass] = (__bridge id)kSecClassGenericPassword;
-    query[(__bridge id)kSecAttrService] = providerIdentifier;
-    query[(__bridge id)kSecAttrAccount] = identityIdentifier;
     
-    OSStatus status = SecItemDelete((__bridge CFDictionaryRef)query);
-    return status == noErr;
+    BOOL success = NO;
+    
+    // normal secret
+    {
+        NSMutableDictionary *query = [[NSMutableDictionary alloc] init];
+        query[(__bridge id)kSecClass] = (__bridge id)kSecClassGenericPassword;
+        query[(__bridge id)kSecAttrService] = providerIdentifier;
+        query[(__bridge id)kSecAttrAccount] = identityIdentifier;
+        
+        OSStatus status = SecItemDelete((__bridge CFDictionaryRef)query);
+        success = status == noErr;
+    }
+    
+    // biometric secret
+    {
+        NSMutableDictionary *query = [[NSMutableDictionary alloc] init];
+        query[(__bridge id)kSecClass] = (__bridge id)kSecClassGenericPassword;
+        query[(__bridge id)kSecAttrService] = providerIdentifier;
+        query[(__bridge id)kSecAttrAccount] = [self biometricAccountValueForIdentifier: identityIdentifier];
+        
+        OSStatus status = SecItemDelete((__bridge CFDictionaryRef)query);
+        success = success || status == noErr;
```

Comment:
Am not familiar with Objective-C, but is this line correct? In the normal secret you use `success = status == noErr;` which I think results in `YES` or `NO`, but this line will always end up as `success = success` and will never trigger the OR-statement.

Reply:
I think I understand, the evaluation happens before the assignment.